File: algorithms/search/binary_search/binary_search.java
class binary_search {
    public static int search(int arr[], int key ) {
        int start = 0;
        int end = arr.length - 1;
        while (start <= end) {
            int mid = start + (end - start) / 2;
            if (arr[mid] == key) {
                return mid;
            } else if (arr[mid] < key) {
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        return -1;
    }

    public static void result(int idx) {
        if (idx == -1) {
            System.out.print("Element not found \n");
        } else {
            System.out.print("Element found at index " + idx + " \n");
        }
    }

    public static void main(String[] args) {
        int arr[] = {-5, 0, 1, 2, 9, 17, 42, 69};
        result(search(arr, 3));
        result(search(arr, 2));
    }
}

File: algorithms/search/binary_search/binary_search.php
<?php

function search($arr, $key){
    $start = 0;
    $end = count($arr)-1;
    while ($start <= $end) {
        $mid = floor(($start + $end) / 2);
        if ($arr[$mid] == $key) {
            return $mid;
        }
        if ($arr[$mid] < $key) {
            $start = $mid + 1;
        } else {
            $end = $mid - 1;
        }
    }
    return -1;
}

function result($idx){
    if($idx == -1){
        echo "Element not found \n";
    } else {
        echo "Element found at index $idx \n";
    }
}

$arr = [-5, 0, 1, 2, 9, 17, 42, 69];
result(search($arr, 3));
result(search($arr, 2));

?>